### PR TITLE
feat: Add cssGridPropertyName to Column and Page for Server Sider Rendering

### PR DESCRIPTION
**Pull request by JamesBurnside**
# Description

Add cssGridPropertyName to Column and Page for Server Sider Rendering

## Motivation & context

#2409
To avoid html mismatches we need the server to be able to specify grid vs ms-grid css property.
Similar to https://github.com/microsoft/fast-dna/pull/2459

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-layouts-react/src/page/page.props.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from "react";
+import { GridDisplay } from "../grid/grid.props";
```
Comment (chrisdholt) [inline packages/fast-layouts-react/src/page/page.props.ts]: @JamesBurnside @nicholasrice I wonder if we want to move this definition *out* from Grid? This change makes the interface a bit more of a utility specific to the CSS grid property name. It no longer is specific to the grid component implementation, and I think we should likely move it as part of this change.

File: packages/fast-layouts-react/src/page/page.props.ts [hunk 2]
```diff
@@ -14,6 +15,12 @@ export interface PageHandledProps extends PageManagedClasses {
      * The max width of the page
      */
     maxWidth?: string;
+
+    /**
+     * The CSS grid display, "grid" when true and "-ms-grid" when false
```
Comment (nicholasrice) [inline packages/fast-layouts-react/src/page/page.props.ts]: I'm not following this comment - what is "true" or "false" here?

Maybe change to "The value assigned to the CSS display property"

File: packages/fast-layouts-react/src/column/column.tsx [hunk 3]
```diff
@@ -170,7 +171,9 @@ export class Column extends Foundation<ColumnHandledProps, ColumnUnhandledProps,
                 : this.props.order;
         }
 
-        const gridStyles: React.CSSProperties = canUseCssGrid()
+        const canUseCssGridStyle: boolean =
+            this.props.cssGridPropertyName === "grid" ? true : canUseCssGrid();
```
Comment (nicholasrice) [inline packages/fast-layouts-react/src/column/column.tsx]: Do you think it's unintuitive that this still has the potential to generate CSS grid properties even if I'm explicitly setting the `cssGridPropertyName` to `"-ms-grid"`. Probably not a common scenario, but I think this should mimic the handled/unhandled pattern of form controls where, if set, `cssGridPropertyName` would always be respected. Thoughts?

File: packages/fast-layouts-react/src/page/page.props.ts (hunk 1 above)
Comment (nicholasrice) [reply packages/fast-layouts-react/src/page/page.props.ts]: I'd be open to moving within this package, but I think it's a bit fragmented of an implementation to put into web-utilities or other packages.

File: packages/fast-layouts-react/src/page/page.props.ts (hunk 2 above)
Comment (JamesBurnside) [reply packages/fast-layouts-react/src/page/page.props.ts]: I'll update to that 👍 (The value assigned to the CSS display property. The property is assigned "grid" when true and "-ms-grid" when false.

File: packages/fast-layouts-react/src/page/page.props.ts (hunk 1 above)
Comment (JamesBurnside) [reply packages/fast-layouts-react/src/page/page.props.ts]: I could move into fast-layouts-react/utilities into its own file of something like "grid-display.ts" or "layout-styles.ts"

File: packages/fast-layouts-react/src/column/column.tsx (hunk 3 above)
Comment (JamesBurnside) [reply packages/fast-layouts-react/src/column/column.tsx]: Yeah that's a fair point, I'll update to

    const canUseCssGridStyle: boolean =
          this.props.cssGridPropertyName === "grid" ? true :
          this.props.cssGridPropertyName === "-ms-grid" ? false : 
          canUseCssGrid();

File: packages/fast-layouts-react/src/page/page.props.ts (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-layouts-react/src/page/page.props.ts]: Yes, this is what I'm referring to. I think we want it within this package, just not in the grid.props.ts file.

I think for now it's likely fine to even just have an index file in a utils folder until we need to add more:
```
fast-layouts-react/utilities/index.ts
```

**Comment by nicholasrice** [comment]
Thanks for the PR @JamesBurnside

**Comment by chrisdholt** [comment]
Merged - thx @JamesBurnside!

**Pull request merged**